delivery/src/main/java/nl/softwarestrijders/waiter/delivery/core/domain/DeliveryAddress.java

```diff
@@ -0,0 +1,26 @@
+package nl.softwarestrijders.waiter.delivery.core.domain;
+
+import nl.softwarestrijders.waiter.delivery.core.common.Generated;
+import nl.softwarestrijders.waiter.delivery.core.domain.exceptions.InvalidHouseNumberException;
+import nl.softwarestrijders.waiter.delivery.core.domain.exceptions.InvalidPostalCodeException;
+import nl.softwarestrijders.waiter.delivery.core.domain.exceptions.InvalidStreetNameException;
+
+import java.util.Objects;
+
+/**
+ * This is the address where the order should be delivered and does not have to be the same as the Customer's address.
+ */
+public record DeliveryAddress(String streetName, int houseNumber, String addition, String postalCode, String city) {
+    public DeliveryAddress {
+        Objects.requireNonNull(streetName);
```

Comment:
```suggestion
        Objects.requireNonNull(streetName, "Need a valid street name to create a new address");
```

You could write a description for this exception throw like you do in other cases where you use `Objects.requireNonNull()`